IPython/terminal/console/completer.py

```diff
@@ -43,7 +48,8 @@ def complete_request(self, text):
         
         msg = self.client.shell_channel.get_msg(timeout=self.timeout)
         if msg['parent_header']['msg_id'] == msg_id:
-            return msg["content"]["matches"]
+            cursor_start = msg['content']['cursor_start']
+            return [ line[:cursor_start] + m for m in msg['content']['matches'] ]
```

Comment:
@minrk, you may want to use what I had started [here](https://github.com/ipython/ipython/pull/6714/files), to account for `cursor_end`.